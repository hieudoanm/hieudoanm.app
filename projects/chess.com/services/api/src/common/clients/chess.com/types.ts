export type ChessPlayer = {
  'avatar': string;
  'player_id': number;
  '@id': string;
  'url': string;
  'name': string;
  'username': string;
  'title': string;
  'followers': number;
  'country': string;
  'location': string;
  'last_online': number;
  'joined': number;
  'status': string;
  'is_streamer': boolean;
  'twitch_url': string;
  'verified': boolean;
  'league': string;
};

export type ChessStatsLast = { rating: number; date: number; rd: number };
export type ChessStatsBest = { rating: number; date: number; game: string };
export type ChessStatsRecord = { win: number; loss: number; draw: number };
export type ChessTactic = { rating: number; date: number };
export type ChessPuzzleRush = { total_attempts: number; score: number };

export type ChessStats = {
  chess_daily: {
    last: ChessStatsLast;
    best: ChessStatsBest;
    record: ChessStatsRecord & {
      time_per_move: number;
      timeout_percent: number;
    };
  };
  chess960_daily: {
    last: ChessStatsLast;
    best: ChessStatsBest;
    record: ChessStatsRecord & {
      time_per_move: number;
      timeout_percent: number;
    };
  };
  chess_rapid: {
    last: ChessStatsLast;
    best: ChessStatsBest;
    record: ChessStatsRecord;
  };
  chess_bullet: {
    last: ChessStatsLast;
    best: ChessStatsBest;
    record: ChessStatsRecord;
  };
  chess_blitz: {
    last: ChessStatsLast;
    best: ChessStatsBest;
    record: ChessStatsRecord;
  };
  fide: number;
  tactics: {
    highest: ChessTactic;
    lowest: ChessTactic;
  };
  puzzle_rush: {
    best: ChessPuzzleRush;
    daily: ChessPuzzleRush;
  };
};

export type ChessFullPlayer = ChessPlayer & { stats: ChessStats } & {
  archives: string[];
};

export type ChessGame = {
  url: string;
  pgn: string;
  time_control: string;
  end_time: number;
  rated: boolean;
  accuracies: {
    white: number;
    black: number;
  };
  tcn: string;
  uuid: string;
  initial_setup: string;
  fen: string;
  time_class: string;
  rules: string;
  white: {
    'rating': number;
    'result': string;
    '@id': string;
    'username': string;
    'uuid': string;
  };
  black: {
    'rating': number;
    'result': string;
    '@id': string;
    'username': string;
    'uuid': string;
  };
};

export type ChessTitle =
  | 'GM'
  | 'WGM'
  | 'IM'
  | 'WIM'
  | 'FM'
  | 'WFM'
  | 'CM'
  | 'WCM'
  | 'NM'
  | 'WNM';
