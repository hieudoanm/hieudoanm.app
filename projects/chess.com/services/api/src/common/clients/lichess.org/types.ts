export type FullCloudEvaluation = {
  fen: string;
  knodes: number;
  depth: number;
  principalVariationSearch: PrincipalVariationSearch[];
};

export type PrincipalVariationSearch = {
  nextMoves: string;
  centipawn: number;
  pawn: number;
};

export type CloudEvaluation = {
  fen: string;
  knodes: number;
  depth: number;
  pvs: PVS[];
};

export type PVS = {
  moves: string;
  cp: number;
};
