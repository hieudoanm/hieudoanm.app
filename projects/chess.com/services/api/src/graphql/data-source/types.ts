import { Player, Title } from '@prisma/client';

export type CountryTotal = {
  countryCode: string;
  country: string;
  total: number;
};

export type Titled = {
  averageRapidRating: number;
  averageBlitzRating: number;
  averageBulletRating: number;
  maxRapidRating: number;
  maxBlitzRating: number;
  maxBulletRating: number;
  total: number;
  players: Player[];
};

export type Country = {
  averageRapidRating: number;
  averageBlitzRating: number;
  averageBulletRating: number;
  maxRapidRating: number;
  maxBlitzRating: number;
  maxBulletRating: number;
  total: number;
  players: Player[];
  titles: Title[];
};

export type GamesSynced = { total: number; synced: number; existed: number; };
