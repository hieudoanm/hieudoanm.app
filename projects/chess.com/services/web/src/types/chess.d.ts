export type Insights = {
  username?: string;
  games?: Games;
  accuracy?: Accuracy;
  results?: Results;
  opponents?: Opponent[];
};

export type Results = {
  win?: Result[];
  draw?: Result[];
  loss?: Result[];
  timeOfDays?: ResultsByTimeOfDay[];
  daysOfWeek?: ResultsByDayOfWeek[];
};

export type Result = {
  result?: string;
  count?: number;
};

export type ResultsByTimeOfDay = {
  win?: number;
  draw?: number;
  loss?: number;
  timeOfDay?: string;
};

export type ResultsByDayOfWeek = {
  win?: number;
  draw?: number;
  loss?: number;
  dayOfWeek?: string;
};

export type Games = {
  total?: number;
  win?: number;
  draw?: number;
  loss?: number;
  periods?: GamesByPeriod[];
  timeOfDays?: GamesByTimeOfDay[];
  daysOfWeek?: GamesByDayOfWeek[];
};

export type GamesByPeriod = { games?: number; period?: number; };
export type GamesByTimeOfDay = { games?: number; timeOfDay?: string; };
export type GamesByDayOfWeek = { games?: number; dayOfWeek?: string; };

export type Accuracy = {
  average?: number;
  win?: number;
  draw?: number;
  loss?: number;
  periods?: AccuracyByPeriod[];
  timeOfDays?: AccuracyByTimeOfDay[];
  daysOfWeek?: AccuracyByDayOfWeek[];
};

export type AccuracyByPeriod = { average?: number; period?: number; };
export type AccuracyByTimeOfDay = { average?: number; timeOfDay?: string; };
export type AccuracyByDayOfWeek = { average?: number; dayOfWeek?: string; };

export type Opponent = {
  opponent?: string;
  games?: number;
  win?: number;
  draw?: number;
  loss?: number;
};
