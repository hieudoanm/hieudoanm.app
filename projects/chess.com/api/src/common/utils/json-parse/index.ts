import { logger } from '../../libs/log';

export const jsonParse = <T>(jsonString: string | null): T | undefined => {
  if (!jsonString) return;
  try {
    return JSON.parse(jsonString);
  } catch (error) {
    logger.error(`jsonParse error=${error}`);
    return {} as T;
  }
};
