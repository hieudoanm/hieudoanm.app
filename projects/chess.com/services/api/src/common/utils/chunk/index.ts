export const chunk = <T>(array: T[], perChunk: number) => {
  const resultArray: T[][] = [];
  for (let index = 0; index < array.length; index += perChunk) {
    const chunkArray: T[] = array.slice(index, index + perChunk);
    resultArray.push(chunkArray);
  }
  return resultArray;
};
