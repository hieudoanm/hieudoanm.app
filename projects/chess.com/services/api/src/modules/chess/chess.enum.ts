import {
  ONE_MONTH,
  ONE_QUARTER,
  ONE_WEEK,
  ONE_YEAR,
} from '../../common/constants';

export type TimeRange = 'WEEK' | 'MONTH' | 'QUARTER' | 'YEAR';

export const timeRangeInMilliseconds: Record<TimeRange, number> = {
  WEEK: ONE_WEEK,
  MONTH: ONE_MONTH,
  QUARTER: ONE_QUARTER,
  YEAR: ONE_YEAR,
};

export const timeRangeInDays: Record<TimeRange, number> = {
  WEEK: 7,
  MONTH: 30,
  QUARTER: 90,
  YEAR: 365,
};
