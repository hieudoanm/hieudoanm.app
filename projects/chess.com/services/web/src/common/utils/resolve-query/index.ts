import { ParsedUrlQuery } from 'node:querystring';

export const resolveQuery = <T>(
  query: ParsedUrlQuery,
  key: string,
  fallback: string = ''
): string => {
  const value: string | string[] | undefined = query[key];
  if (!value) return fallback;
  if (typeof value === 'string') return value;
  return value[0];
};
