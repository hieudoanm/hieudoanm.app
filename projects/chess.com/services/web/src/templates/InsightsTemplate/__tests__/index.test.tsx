import { render } from '@testing-library/react';
import { InsightsTemplate } from '..';
import { Insights } from '@chess/types/chess';
import { mockResizeObserver } from '@chess/common/utils/mock-resize-observer';

jest.mock('next/router', () => ({
  useRouter: jest
    .fn()
    .mockReturnValue({ asPath: '', events: { on: jest.fn(), off: jest.fn() } }),
}));

describe('InsightsTemplate', () => {
  beforeEach(() => {
    mockResizeObserver();
  });

  it('to match snapshot', () => {
    const { container } = render(
      <InsightsTemplate insights={{} as Insights} />
    );
    expect(container).toMatchSnapshot();
  });
});
