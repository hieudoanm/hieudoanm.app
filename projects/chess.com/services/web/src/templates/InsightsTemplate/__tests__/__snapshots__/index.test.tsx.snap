// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsTemplate to match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4 md:gap-y-8"
  >
    <div
      class="chakra-card border border-gray-200 css-2dlz7"
    >
      <div
        class="chakra-card__header css-0"
      >
        <h1
          class="text-xl md:text-3xl"
        >
          Overview
        </h1>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex flex-col gap-y-2"
            >
              <p
                class="text-base md:text-lg"
              >
                Games played
              </p>
              <p
                class="text-lg md:text-xl"
              >
                <b>
                  0
                </b>
              </p>
            </div>
            <div
              class="flex items-center gap-x-2 md:gap-x-4"
            >
              <div
                class="flex flex-col gap-y-1"
              >
                <div
                  class="flex items-center gap-x-1 md:gap-x-2"
                >
                  <svg
                    class="chakra-icon css-g37jgb"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                    />
                  </svg>
                  <b
                    class="text-base md:text-lg"
                  >
                    0.00
                    %
                  </b>
                </div>
                <div
                  class="text-right text-xs capitalize md:text-sm"
                >
                  0
                   
                  won
                </div>
              </div>
              <div
                class="flex flex-col gap-y-1"
              >
                <div
                  class="flex items-center gap-x-1 md:gap-x-2"
                >
                  <svg
                    class="chakra-icon css-15t4wcl"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    />
                  </svg>
                  <b
                    class="text-base md:text-lg"
                  >
                    0.00
                    %
                  </b>
                </div>
                <div
                  class="text-right text-xs capitalize md:text-sm"
                >
                  0
                   
                  drawn
                </div>
              </div>
              <div
                class="flex flex-col gap-y-1"
              >
                <div
                  class="flex items-center gap-x-1 md:gap-x-2"
                >
                  <svg
                    class="chakra-icon css-f11o2j"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
                    />
                  </svg>
                  <b
                    class="text-base md:text-lg"
                  >
                    0.00
                    %
                  </b>
                </div>
                <div
                  class="text-right text-xs capitalize md:text-sm"
                >
                  0
                   
                  lost
                </div>
              </div>
            </div>
          </div>
          <div
            class="aspect-video"
          >
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 100%; min-width: 0;"
            />
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex flex-col gap-y-2"
            >
              <p
                class="text-base md:text-lg"
              >
                Average accuracy
              </p>
              <p
                class="text-lg md:text-xl"
              >
                <b>
                  0
                </b>
              </p>
            </div>
            <div
              class="flex items-center gap-x-2 md:gap-x-4"
            >
              <div
                class="flex flex-col gap-y-1"
              >
                <p
                  class="text-right text-xs capitalize md:text-sm"
                >
                  When you win
                </p>
                <div
                  class="flex items-center gap-x-1 md:gap-x-2"
                >
                  <svg
                    class="chakra-icon css-g37jgb"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                    />
                  </svg>
                  <b
                    class="text-base md:text-lg"
                  >
                    0.00
                    %
                  </b>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-1"
              >
                <p
                  class="text-right text-xs capitalize md:text-sm"
                >
                  When you draw
                </p>
                <div
                  class="flex items-center gap-x-1 md:gap-x-2"
                >
                  <svg
                    class="chakra-icon css-15t4wcl"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    />
                  </svg>
                  <b
                    class="text-base md:text-lg"
                  >
                    0.00
                    %
                  </b>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-1"
              >
                <p
                  class="text-right text-xs capitalize md:text-sm"
                >
                  When you lose
                </p>
                <div
                  class="flex items-center gap-x-1 md:gap-x-2"
                >
                  <svg
                    class="chakra-icon css-f11o2j"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
                    />
                  </svg>
                  <b
                    class="text-base md:text-lg"
                  >
                    0.00
                    %
                  </b>
                </div>
              </div>
            </div>
          </div>
          <div
            class="aspect-video"
          >
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 100%; min-width: 0;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-card border border-gray-200 css-2dlz7"
    >
      <div
        class="chakra-card__header css-0"
      >
        <h1
          class="text-xl md:text-3xl"
        >
          Game Results
        </h1>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Games you won by ...
          </h2>
          <div
            class="rounded-2xl bg-gray-100 p-8"
          >
            <div
              class="aspect-video"
            >
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 100%; min-width: 0;"
              />
            </div>
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Games you drawn by ...
          </h2>
          <div
            class="rounded-2xl bg-gray-100 p-8"
          >
            <div
              class="aspect-video"
            >
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 100%; min-width: 0;"
              />
            </div>
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Games you lost by ...
          </h2>
          <div
            class="rounded-2xl bg-gray-100 p-8"
          >
            <div
              class="aspect-video"
            >
              <div
                class="recharts-responsive-container"
                style="width: 100%; height: 100%; min-width: 0;"
              />
            </div>
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      />
    </div>
    <div
      class="chakra-card border border-gray-200 css-2dlz7"
    >
      <div
        class="chakra-card__header css-0"
      >
        <h1
          class="text-xl md:text-3xl"
        >
          Time of Day
        </h1>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Games by Time of Day
          </h2>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Accuracy by Time of Day
          </h2>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Results by Time of Day
          </h2>
        </div>
      </div>
    </div>
    <div
      class="chakra-card border border-gray-200 css-2dlz7"
    >
      <div
        class="chakra-card__header css-0"
      >
        <h1
          class="text-xl md:text-3xl"
        >
          Day of Week
        </h1>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Games by Days of Week
          </h2>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Accuracy by Days of Week
          </h2>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="flex flex-col gap-y-4"
        >
          <h2
            class="font-bold"
          >
            Results by Days of Week
          </h2>
        </div>
      </div>
    </div>
    <div
      class="chakra-card border border-gray-200 css-2dlz7"
    >
      <div
        class="chakra-card__header css-0"
      >
        <h1
          class="text-xl md:text-3xl"
        >
          Opponents
        </h1>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-card__body css-0"
      >
        <div
          class="chakra-table__container css-zipzvv"
        >
          <table
            class="chakra-table css-0"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-0"
              >
                <th
                  class="css-0"
                  data-is-numeric="true"
                >
                  No
                </th>
                <th
                  class="css-0"
                >
                  Opponent
                </th>
                <th
                  class="css-0"
                  data-is-numeric="true"
                >
                  <div
                    class="cursor-pointer capitalize"
                  >
                    Games
                  </div>
                </th>
                <th
                  class="css-0"
                  data-is-numeric="true"
                >
                  <div
                    class="flex items-center justify-end gap-2"
                  >
                    <div
                      class="cursor-pointer capitalize"
                    >
                      Win
                    </div>
                    <div
                      class="cursor-pointer capitalize"
                    >
                      (%)
                    </div>
                  </div>
                </th>
                <th
                  class="css-0"
                  data-is-numeric="true"
                >
                  <div
                    class="flex items-center justify-end gap-2"
                  >
                    <div
                      class="cursor-pointer capitalize"
                    >
                      Draw
                    </div>
                    <div
                      class="cursor-pointer capitalize"
                    >
                      (%)
                    </div>
                  </div>
                </th>
                <th
                  class="css-0"
                  data-is-numeric="true"
                >
                  <div
                    class="flex items-center justify-end gap-2"
                  >
                    <div
                      class="cursor-pointer capitalize"
                    >
                      Loss
                    </div>
                    <div
                      class="cursor-pointer capitalize"
                    >
                      (%)
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="css-0"
            />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
